src/core/particle_data.hpp

```diff
@@ -1028,6 +1035,6 @@ void pointer_to_swimming(Particle const *p,
 void pointer_to_rotational_inertia(Particle const *p, double const *&res);
 #endif
 
-bool particle_exists(int part);
+bool particle_exists(int part) noexcept;
```

Comment:
The `noexcept` is a lie, as `particle_exists` potentially calls `build_particle_node`, which can throw for various reasons.